.github/scripts/create-e2e-test-report.ts

```diff
@@ -145,63 +84,125 @@ async function main() {
           };
           testSuite.testCases.push(testCase);
         }
+
+        const testFile: TestFile = {
+          path: normalizeTestPath(suite.$.file),
+          tests: testSuite.tests,
+          passed: testSuite.passed,
+          failed: testSuite.failed,
+          skipped: testSuite.skipped,
+          time: testSuite.time,
+          testSuites: [testSuite],
+        };
+
         const testRun: TestRun = {
           // regex to remove the shard number from the job name
           name: testSuite.job.name.replace(/\s+\(\d+\)$/, ''),
-          testSuites: [testSuite],
+          testFiles: [testFile],
         };
+
         const existingRun = testRuns.find((run) => run.name === testRun.name);
         if (existingRun) {
-          existingRun.testSuites.push(testSuite);
+          const existingFile = existingRun.testFiles.find(
+            (file) => file.path === testFile.path,
+          );
+          if (existingFile) {
+            existingFile.testSuites.push(testSuite);
+          } else {
+            existingRun.testFiles.push(testFile);
+          }
         } else {
           testRuns.push(testRun);
         }
       }
     }
 
     for (const testRun of testRuns) {
-      const deduped: { [path: string]: TestSuite } = {};
-      for (const suite of testRun.testSuites) {
-        const existing = deduped[suite.path];
-        // If there is a duplicate, we keep the suite with the latest date
-        if (!existing || existing.date < suite.date) {
-          deduped[suite.path] = suite;
+      for (const testFile of testRun.testFiles) {
+        // Group test suites by name
+        const suitesByName: Record<string, TestSuite[]> = {};
+
+        for (const suite of testFile.testSuites) {
+          if (!suitesByName[suite.name]) {
+            suitesByName[suite.name] = [];
+          }
+          suitesByName[suite.name].push(suite);
         }
-      }
 
-      const suites = Object.values(deduped);
+        // Determine the latest test suite by date and nest attempts
+        const attempts: TestSuite[][] = [];
+        for (const suites of Object.values(suitesByName)) {
+          suites.sort((a, b) => b.date.getTime() - a.date.getTime()); // sort newest first
+          const [latest, ...otherAttempts] = suites;
+          latest.attempts = otherAttempts;
+          attempts.push(otherAttempts);
+        }
 
-      const title = `<strong>${testRun.name}</strong>`;
-      console.log(consoleBold(title));
+        // Remove the nested attempts from the top-level list
+        const attemptSet = new Set(attempts.flat().map((s) => s));
```

Comment:
Does the `.map((s) => s)` need to be here?

```suggestion
        const attemptSet = new Set(attempts.flat());
```